connectors/src/lib/data_sources.ts

```diff
@@ -342,3 +313,37 @@ export function renderMarkdownSection(
 
   return top;
 }
+
+// Will render the document based on title, optional createdAt and updatedAt and a structured
+// content. The title, createdAt and updatedAt will be presented in a standardized way across
+// connectors. The title should not include any `\n`.
+// If the title is too long it will be truncated and the remainder of the title will be set as
+// content of the top-level section.
+export function renderDocumentForTitleAndContent({
+  title,
+  createdAt,
+  updatedAt,
+  content,
+}: {
+  title: string | null;
+  createdAt?: Date;
+  updatedAt?: Date;
+  content: CoreAPIDataSourceDocumentSection | null;
+}): CoreAPIDataSourceDocumentSection {
+  if (title && title.trim()) {
+    title = `$title: ${title}\n`;
+  } else {
+    title = null;
+  }
+  const c = renderPrefixSection(title);
+  if (createdAt) {
```

Comment:
I don't want to call renderPrefix on structured data. It should be in or out consistently.

I agree that it grows a bit but this is somewhat important information. I'll use 192 instead of 256 for the prefix

Reply:
(done)